plugins/woocommerce/includes/wc-core-functions.php

```diff
@@ -2532,7 +2532,7 @@ function wc_selected( $value, $options ) {
 function wc_get_server_database_version() {
 	global $wpdb;
 
-	if ( empty( $wpdb->is_mysql ) || ! $wpdb->use_mysqli ) {
+	if ( ! property_exists( $wpdb, 'is_mysql' ) || empty( $wpdb->is_mysql ) || ! property_exists( $wpdb, 'use_mysqli' ) || ! $wpdb->use_mysqli ) {
```

Comment:
I *think* this is functionally the same as this slightly shorter form (because `empty` will not complain if the var does not exist, and because if `use_mysqli` *is* set it acts as a proxy for `! $wpdb->use_mysqli`):

```suggestion
	if ( empty( $wpdb->is_mysql ) || empty( $wpdb->use_mysqli ) ) {
```

(No need to change, though, if we prefer to explicitness of the existing approach.)

Reply:
Nice -- I'll use your solution. If it's a problem the test will tell us. ☺️

Reply:
Just saw this PR after searching for the undefined error in my PR.

I've looked into it a bit and it looks like `use_mysqli` var is a private var. See https://github.com/WordPress/WordPress/blob/master/wp-includes/class-wpdb.php#L706

`empty($wpdb->use_mysqli)` will always return true in that case.